internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,293 @@
+		return errors.New("bundle image cannot be empty")
+	}
+	if len(rp.DBPath) == 0 {
+		return errors.New("registry database path cannot be empty")
+	}
+
+	if len(rp.Namespace) == 0 {
+		return errors.New("pod namespace cannot be empty")
+	}
+
+	if len(rp.BundleAddMode) == 0 {
+		return errors.New("bundle add mode cannot be empty")
+	}
+	if rp.BundleAddMode != semverBundleAddMode && rp.BundleAddMode != replacesBundleAddMode {
+		return errors.New("invalid bundle mode")
+	}
+
+	return nil
+}
+
+// getPodName will return a string constructed from the bundle Image name
+func getPodName(bundleImage string) (string, error) {
+	image, err := container.ImageFromString(bundleImage)
+	if err != nil {
+		return "", fmt.Errorf("invalid bundle image name: %s", bundleImage)
+	}
+
+	return image.Name + "-index", nil
+}
+
+// podForBundleRegistry constructs and returns the registry pod definition
+func (rp *RegistryPod) podForBundleRegistry() (*corev1.Pod, error) {
+	podName, err := getPodName(rp.BundleImage)
+	if err != nil {
+		return nil, fmt.Errorf("error in building pod: %v", err)
+	}
+
+	containerCmd, err := rp.getContainerCmd()
+	if err != nil {
+		return nil, fmt.Errorf("error in parsing container command: %v", err)
+	}
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: rp.Namespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  defaultContainerName,
+					Image: rp.IndexImage,
+					Command: []string{
+						"/bin/sh",
+						"-c",
+						containerCmd,
+					},
+					Ports: []corev1.ContainerPort{
+						{Name: defaultContainerPortName, ContainerPort: rp.GRPCPort},
+					},
+				},
+			},
+		},
+	}
+
+	return pod, nil
+}
+
+// getContainerCmd uses templating to construct the container command
+func (rp *RegistryPod) getContainerCmd() (string, error) {
+	const containerCommand = "/bin/mkdir -p {{ .DBPath | basename }} &&" +
+		"/bin/opm registry add -d {{ .DBPath | basename }} -b {{.BundleImage}} --mode={{.BundleAddMode}} &&" +
+		"/bin/opm registry serve -d {{ .DBPath | basename }} -p {{.GRPCPort}}"
+	type bundleCmd struct {
+		BundleImage, DBPath, BundleAddMode string
+		GRPCPort                           int32
+	}
+
+	var command = bundleCmd{rp.BundleImage, rp.DBPath, rp.BundleAddMode, rp.GRPCPort}
+
+	out := &bytes.Buffer{}
+
+	// create a custom basename template function
+	funcMap := template.FuncMap{
+		"basename": filepath.Base,
+	}
+
+	// add the custom basename template function to the
+	// template's FuncMap and parse the containerCommand
+	tmp := template.Must(template.New("containerCommand").Funcs(funcMap).Parse(containerCommand))
+
+	err := tmp.Execute(out, command)
+	if err != nil {
+		return "", fmt.Errorf("error in parsing container command: %w", err)
+	}
+
+	return out.String(), nil
+}
+
+// FetchPodLogs gets the logs from the registry pod
+func (rp *RegistryPod) FetchPodLogs(ctx context.Context, pod *corev1.Pod) (string, error) {
```

Comment:
```suggestion
func (rp *RegistryPod) GetLogs(ctx context.Context, pod *corev1.Pod) (string, error) {
```

Reply:
I agree the pod can be an attribute of the structure. I think I suggested it in another comment as well.

Reply:
good point @estroz and @camilamacedo86 
Added pod as an unexported field to `RegistryPod` struct